README: add chapter describing NVDIMM pre-provisioning with ipmctl

okartau opened the pull request:
Resolves #217

pohly commented on README.md:
````diff
@@ -375,10 +376,30 @@ development and testing can be done using QEMU-emulated persistent
+Example of creating non-interleaved goal using all NVDIMMs:
+```sh
+# ipmctl create -goal PersistentMemoryType=AppDirectNotInterleaved
+```
````
Ah, so "goal" is a special term in ipmctl. I didn't know that and other readers might have the same problem. Let's write our README so that as little prior knowledge is needed as possible.

We can (and have to) use it here of course, just not in plain text.

pohly commented on README.md:
````diff
@@ -375,10 +376,30 @@ development and testing can be done using QEMU-emulated persistent
+# ipmctl create -goal PersistentMemoryType=AppDirectNotInterleaved
+```
+
+Multiple NVDIMMs can be combined using interleaving creation goals.
````
"Alternatively, multiple NVDIMMs can be interleaved."

pohly commented on README.md:
```diff
@@ -375,10 +376,30 @@ development and testing can be done using QEMU-emulated persistent
+Multiple NVDIMMs can be combined using interleaving creation goals.
+This causes the data to be striped over multiple NVDIMM devices
+and allows one region to be larger than single NVDIMM, thus enabling
+creation of PMEM-CSI volumes that larger than single NVDIMM.
```
"Performance is also higher." - right?

okartau commented:
Pushed new version. I avoid using word "goal" in text so that we dont have to explain it explicitly. It remains visible in impctl command examples, where it is (hopefully) self-explaining (or user needs to check ipmctl help/manual). 
I added sentence stating App Direct Mode has two options (borrowed that sentence from Optane config description).I realized we actually dont mention anywhere that PMEM-CSI driver needs Optane devices to be configured in App Direct mode, so hopefully that sentence helps to clarify that as well.
I re-worded some text to be shorter and avoid repetition, and added note about perf. improvement that interleaved set brings. I tried to cover all of above comments areas.

avalluri commented:
> Looks good to me.
> 
> @avalluri if you agree, can you merge it?

Yes, let's merge it.
